projects/control-service/projects/pipelines_control_service/src/integration-test/java/com/vmware/taurus/datajobs/it/common/WebHookServerMockExtension.java

```diff
@@ -0,0 +1,257 @@
+/*
+ * Copyright 2021 VMware, Inc.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.vmware.taurus.datajobs.it.common;
+
+import static com.vmware.taurus.datajobs.it.common.BaseIT.NEW_TEST_TEAM_NAME;
+import static com.vmware.taurus.datajobs.it.common.BaseIT.TEST_CLIENT_ERROR_JOB_NAME;
+import static com.vmware.taurus.datajobs.it.common.BaseIT.TEST_CLIENT_ERROR_TEAM;
+import static com.vmware.taurus.datajobs.it.common.BaseIT.TEST_INTERNAL_ERROR_JOB_NAME;
+import static com.vmware.taurus.datajobs.it.common.BaseIT.TEST_INTERNAL_ERROR_RETRIED_JOB_NAME;
+import static com.vmware.taurus.datajobs.it.common.BaseIT.TEST_INTERNAL_ERROR_RETRIED_TEAM;
+import static com.vmware.taurus.datajobs.it.common.BaseIT.TEST_INTERNAL_ERROR_TEAM;
+import static com.vmware.taurus.datajobs.it.common.BaseIT.TEST_JOB_1;
+import static com.vmware.taurus.datajobs.it.common.BaseIT.TEST_JOB_2;
+import static com.vmware.taurus.datajobs.it.common.BaseIT.TEST_JOB_3;
+import static com.vmware.taurus.datajobs.it.common.BaseIT.TEST_JOB_4;
+import static com.vmware.taurus.datajobs.it.common.BaseIT.TEST_JOB_5;
+import static com.vmware.taurus.datajobs.it.common.BaseIT.TEST_JOB_6;
+import static com.vmware.taurus.datajobs.it.common.BaseIT.TEST_JOB_NAME;
+import static com.vmware.taurus.datajobs.it.common.BaseIT.TEST_TEAM_NAME;
+import static org.junit.jupiter.api.extension.ExtensionContext.Namespace.GLOBAL;
+import static org.mockserver.matchers.Times.exactly;
+import static org.mockserver.model.HttpRequest.request;
+import static org.mockserver.model.HttpResponse.response;
+
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+import io.netty.handler.codec.http.HttpMethod;
+import org.junit.jupiter.api.extension.BeforeAllCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.mockserver.client.server.MockServerClient;
+import org.mockserver.integration.ClientAndServer;
+import org.mockserver.model.Header;
+import org.mockserver.model.HttpResponse;
+import org.mockserver.model.HttpStatusCode;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+public class WebHookServerMockExtension implements BeforeAllCallback, ExtensionContext.Store.CloseableResource {
```

Comment:
Please add javadoc on what it does and how it is used. 

Look at the excellent MockitoExtension documentation for inspiration: https://www.javadoc.io/static/org.mockito/mockito-junit-jupiter/4.0.0/org/mockito/junit/jupiter/MockitoExtension.html